pulumi convert panics with concurrent map write

### What happened?

Concurrent map access in
https://github.com/pulumi/pulumi/blob/a4d1a24984c2083e9a6dc964a2e74d9649bf19fa/pkg/codegen/convert/caching_mapper.go#L37

Caught by pre-testing bridge release candidate on cloud-ampq.



### Example

Here's the output.

https://github.com/pulumi/pulumi-cloudamqp/actions/runs/13573399853/job/37943756080?pr=636#step:7:218

```
(cd provider && go build  -o /home/runner/work/pulumi-cloudamqp/pulumi-cloudamqp/bin/pulumi-tfgen-cloudamqp -ldflags "-X github.com/pulumi/pulumi-cloudamqp/provider/v3/pkg/version.Version=3.20.4-alpha.1740681144+cb6c0f0 " github.com/pulumi/pulumi-cloudamqp/provider/v3/cmd/pulumi-tfgen-cloudamqp)
go: downloading github.com/pulumi/pulumi-terraform-bridge/v3 v3.103.1-0.20250227182354-ebf4276b8dd1
go: downloading github.com/cloudamqp/terraform-provider-cloudamqp v1.32.2
go: downloading github.com/pulumi/pulumi/sdk/v3 v3.152.0
go: downloading github.com/pulumi/terraform-plugin-sdk/v2 v2.0.0-20250221232320-8d4cfd37a3cd
go: downloading github.com/hashicorp/terraform-plugin-sdk v1.17.2
go: downloading github.com/blang/semver v3.5.1+incompatible
go: downloading github.com/golang/glog v1.2.2
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/hashicorp/hcl/v2 v2.23.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/pulumi/inflector v0.1.1
go: downloading github.com/pulumi/pulumi-java/pkg v0.19.0
go: downloading github.com/pulumi/pulumi-yaml v1.12.0
go: downloading github.com/pulumi/pulumi/pkg/v3 v3.152.0
go: downloading github.com/pulumi/schema-tools v0.1.2
go: downloading github.com/spf13/afero v1.10.0
go: downloading github.com/russross/blackfriday/v2 v2.1.0
go: downloading github.com/spf13/cobra v1.8.0
go: downloading github.com/teekennedy/goldmark-markdown v0.3.0
go: downloading github.com/yuin/goldmark v1.7.4
go: downloading golang.org/x/mod v0.22.0
go: downloading golang.org/x/text v0.22.0
go: downloading github.com/Masterminds/semver v1.5.0
go: downloading github.com/agext/levenshtein v1.2.3
go: downloading github.com/golang/protobuf v1.5.4
go: downloading github.com/hashicorp/go-cty v1.4.1-0.20200414143053-d3edf31b6320
go: downloading github.com/opentracing/opentracing-go v1.2.0
go: downloading golang.org/x/net v0.34.0
go: downloading google.golang.org/grpc v1.69.4
go: downloading google.golang.org/protobuf v1.36.3
go: downloading github.com/hashicorp/terraform-plugin-go v0.26.0
go: downloading github.com/hashicorp/terraform-plugin-log v0.9.0
go: downloading github.com/mitchellh/go-testing-interface v1.14.1
go: downloading github.com/pulumi/terraform-diff-reader v0.0.2
go: downloading github.com/zclconf/go-cty v1.16.2
go: downloading github.com/dghubble/sling v1.4.2
go: downloading github.com/google/go-cmp v0.6.0
go: downloading github.com/mitchellh/copystructure v1.2.0
go: downloading github.com/mitchellh/mapstructure v1.5.0
go: downloading github.com/hashicorp/go-uuid v1.0.3
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/hashicorp/hil v0.0.0-20190212132231-97b3a9cdfa93
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading github.com/segmentio/encoding v0.3.5
go: downloading github.com/apparentlymart/go-textseg/v15 v15.0.0
go: downloading github.com/mitchellh/go-wordwrap v1.0.1
go: downloading github.com/ettle/strcase v0.1.1
go: downloading github.com/iancoleman/strcase v0.3.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56
go: downloading github.com/pgavlin/goldmark v1.1.33-0.20200616210433-b5eb04559386
go: downloading github.com/BurntSushi/toml v1.2.1
go: downloading github.com/deckarep/golang-set/v2 v2.5.0
go: downloading github.com/pulumi/esc v0.10.0
go: downloading github.com/grpc-ecosystem/grpc-opentracing v0.0.0-20180507213350-8e809c8a8645
go: downloading github.com/charmbracelet/bubbles v0.16.1
go: downloading github.com/charmbracelet/bubbletea v0.25.0
go: downloading github.com/charmbracelet/lipgloss v0.7.1
go: downloading github.com/pulumi/appdash v0.0.0-20231130102222-75f619a67231
go: downloading github.com/rivo/uniseg v0.4.4
go: downloading github.com/uber/jaeger-client-go v2.30.0+incompatible
go: downloading golang.org/x/sys v0.30.0
go: downloading golang.org/x/term v0.29.0
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/djherbis/times v1.5.0
go: downloading github.com/go-git/go-git/v5 v5.13.1
go: downloading github.com/pgavlin/fx v0.1.6
go: downloading github.com/rogpeppe/go-internal v1.12.0
go: downloading github.com/santhosh-tekuri/jsonschema/v5 v5.0.0
go: downloading github.com/texttheater/golang-levenshtein v1.0.1
go: downloading github.com/edsrzf/mmap-go v1.1.0
go: downloading github.com/natefinch/atomic v1.0.1
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/hashicorp/go-hclog v1.6.3
go: downloading lukechampine.com/frand v1.4.2
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20241015192408-796eee8c2d53
go: downloading github.com/vmihailenco/msgpack v4.0.4+incompatible
go: downloading google.golang.org/genproto v0.0.0-20240311173647-c811ad7063a7
go: downloading github.com/vmihailenco/msgpack/v5 v5.4.1
go: downloading github.com/hashicorp/logutils v1.0.0
go: downloading github.com/mitchellh/reflectwalk v1.0.2
go: downloading github.com/google/go-querystring v1.1.0
go: downloading github.com/hashicorp/go-version v1.7.0
go: downloading github.com/hashicorp/terraform-svchost v0.1.1
go: downloading github.com/hashicorp/go-getter v1.7.5
go: downloading github.com/mitchellh/cli v1.1.5
go: downloading github.com/apparentlymart/go-cidr v1.1.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/mitchellh/hashstructure v1.0.0
go: downloading golang.org/x/crypto v0.33.0
go: downloading github.com/mattn/go-runewidth v0.0.15
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading github.com/atotto/clipboard v0.1.4
go: downloading github.com/containerd/console v1.0.4-0.20230313162750-1ae8d489ac81
go: downloading github.com/mattn/go-isatty v0.0.20
go: downloading github.com/mattn/go-localereader v0.0.1
go: downloading github.com/muesli/ansi v0.0.0-20230316100256-276c6243b2f6
go: downloading github.com/muesli/cancelreader v0.2.2
go: downloading github.com/muesli/reflow v0.3.0
go: downloading github.com/muesli/termenv v0.15.2
go: downloading golang.org/x/sync v0.11.0
go: downloading github.com/pkg/term v1.1.0
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/opentracing/basictracer-go v1.1.0
go: downloading github.com/segmentio/asm v1.1.3
go: downloading github.com/uber/jaeger-lib v2.4.1+incompatible
go: downloading go.uber.org/atomic v1.9.0
go: downloading github.com/sabhiram/go-gitignore v0.0.0-20210923224102-525f6e181f06
go: downloading github.com/kevinburke/ssh_config v1.2.0
go: downloading dario.cat/mergo v1.0.0
go: downloading github.com/ProtonMail/go-crypto v1.1.3
go: downloading github.com/go-git/go-billy/v5 v5.6.1
go: downloading github.com/sergi/go-diff v1.3.2-0.20230802210424-5b0b94c5c0d3
go: downloading github.com/aead/chacha20 v0.0.0-20180709150244-8b13a72661da
go: downloading github.com/fatih/color v1.16.0
go: downloading github.com/vmihailenco/tagparser/v2 v2.0.0
go: downloading github.com/Masterminds/sprig/v3 v3.2.3
go: downloading github.com/armon/go-radix v1.0.0
go: downloading github.com/bgentry/speakeasy v0.1.0
go: downloading github.com/posener/complete v1.2.3
go: downloading cloud.google.com/go/storage v1.39.1
go: downloading github.com/aws/aws-sdk-go v1.50.36
go: downloading cloud.google.com/go v0.112.1
go: downloading github.com/bgentry/go-netrc v0.0.0-20140422174119-9fd32a8b3d3d
go: downloading github.com/hashicorp/go-cleanhttp v0.5.2
go: downloading github.com/klauspost/compress v1.16.7
go: downloading github.com/hashicorp/go-safetemp v1.0.0
go: downloading github.com/ulikunitz/xz v0.5.11
go: downloading golang.org/x/oauth2 v0.23.0
go: downloading google.golang.org/api v0.169.0
go: downloading github.com/google/uuid v1.6.0
go: downloading github.com/spf13/cast v1.5.0
go: downloading github.com/aymanbagabas/go-osc52/v2 v2.0.1
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading github.com/skeema/knownhosts v1.3.0
go: downloading github.com/xanzy/ssh-agent v0.3.3
go: downloading github.com/pjbgf/sha1cd v0.3.0
go: downloading github.com/go-git/gcfg v1.5.1-0.20230307220236-3a3c6141e376
go: downloading github.com/emirpasic/gods v1.18.1
go: downloading github.com/jbenet/go-context v0.0.0-20150711004518-d14ea06fba99
go: downloading github.com/cyphar/filepath-securejoin v0.3.6
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading github.com/Masterminds/goutils v1.1.1
go: downloading github.com/Masterminds/semver/v3 v3.2.0
go: downloading github.com/huandu/xstrings v1.3.3
go: downloading github.com/imdario/mergo v0.3.15
go: downloading github.com/shopspring/decimal v1.3.1
go: downloading cloud.google.com/go/compute/metadata v0.5.2
go: downloading cloud.google.com/go/iam v1.1.6
go: downloading github.com/googleapis/gax-go/v2 v2.12.2
go: downloading github.com/cloudflare/circl v1.3.7
go: downloading gopkg.in/warnings.v0 v0.1.2
go: downloading google.golang.org/genproto/googleapis/api v0.0.0-20241015192408-796eee8c2d53
go: downloading go.opencensus.io v0.24.0
go: downloading go.opentelemetry.io/otel v1.31.0
go: downloading go.opentelemetry.io/otel/trace v1.31.0
go: downloading go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.49.0
go: downloading github.com/jmespath/go-jmespath v0.4.0
go: downloading github.com/go-logr/logr v1.4.2
go: downloading go.opentelemetry.io/otel/metric v1.31.0
go: downloading go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.49.0
go: downloading golang.org/x/time v0.5.0
go: downloading github.com/google/s2a-go v0.1.7
go: downloading github.com/googleapis/enterprise-certificate-proxy v0.3.2
go: downloading github.com/go-logr/stdr v1.2.2
go: downloading github.com/felixge/httpsnoop v1.0.4
/home/runner/work/pulumi-cloudamqp/pulumi-cloudamqp/bin/pulumi-tfgen-cloudamqp schema --out provider/cmd/pulumi-resource-cloudamqp
2025/02/27 18:34:22 Terraform-Provider-CloudAMQP Version: 
warning: Found <elided> in description for [cloudamqp_account_action]. The description and any examples will be dropped in the Pulumi provider.
warning: Found <elided> in description for [cloudamqp_alarm]. The description and any examples will be dropped in the Pulumi provider.
warning: Found <elided> in description for [cloudamqp_plugin]. The description and any examples will be dropped in the Pulumi provider.
warning: Found <elided> in description for [cloudamqp_plugin_community]. The description and any examples will be dropped in the Pulumi provider.
warning: Found <elided> in description for [cloudamqp_rabbitmq_configuration]. The description and any examples will be dropped in the Pulumi provider.
warning: Found <elided> in description for [cloudamqp_security_firewall]. The description and any examples will be dropped in the Pulumi provider.
warning: Found <elided> in description for [cloudamqp_vpc_peering]. The description and any examples will be dropped in the Pulumi provider.
warning: Found <elided> in description for [cloudamqp_webhook]. The description and any examples will be dropped in the Pulumi provider.
panic: fatal: error An assertion has failed: bulk converting examples failed. source error: 
######
  pulumi convert failed
  minimal repro: /tmp/convert-examples-repro3947055978
  full error below
######
convertViaPulumiCLI: pulumi command failed: exit status 2
Stdout:


Stderr:
Converting from terraform...
fatal error: concurrent map writes

goroutine 45 [running]:
github.com/pulumi/pulumi/pkg/v3/codegen/convert.(*cachingMapper).GetMapping(0xc00218d758, {0x33b1690, 0xc0022cb290}, {0xc002298a60, 0x9}, 0xc001df04f8)
	github.com/pulumi/pulumi/pkg/v3/codegen/convert/caching_mapper.go:51 +0xb6
github.com/pulumi/pulumi/pkg/v3/codegen/convert.(*mapperServer).GetMapping(0xc000a0e100, {0x33b1690, 0xc0022cb290}, 0xc000838a50)
	github.com/pulumi/pulumi/pkg/v3/codegen/convert/mapper_server.go:62 +0x366
github.com/pulumi/pulumi/sdk/v3/proto/go/codegen._Mapper_GetMapping_Handler.func1({0x33b1690?, 0xc0022cb290?}, {0x2a30860?, 0xc000838a50?})
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/proto/go/codegen/mapper_grpc.pb.go:96 +0xcb
github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x33b1690, 0xc0022cafc0}, {0x2a30860, 0xc000838a50}, 0xc00220fae0, 0xc001df04e0)
	github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x3c4
github.com/pulumi/pulumi/sdk/v3/proto/go/codegen._Mapper_GetMapping_Handler({0x280d400, 0xc000a0e100}, {0x33b1690, 0xc0022cafc0}, 0xc000810e80, 0xc000a0e140)
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/proto/go/codegen/mapper_grpc.pb.go:98 +0x143
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001dea000, {0x33b1690, 0xc001e250b0}, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef0e0, 0xc001e24750, 0x49cc0f0, 0x0)
	google.golang.org/grpc@v1.67.1/server.go:1394 +0xe2b
google.golang.org/grpc.(*Server).handleStream(0xc001dea000, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef0e0)
	google.golang.org/grpc@v1.67.1/server.go:1805 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.67.1/server.go:1029 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 44
	google.golang.org/grpc@v1.67.1/server.go:1040 +0x125

goroutine 1 [select]:
google.golang.org/grpc/internal/transport.(*Stream).waitOnHeader(0xc0003d9200)
	google.golang.org/grpc@v1.67.1/internal/transport/transport.go:378 +0x7c
google.golang.org/grpc/internal/transport.(*Stream).RecvCompress(...)
	google.golang.org/grpc@v1.67.1/internal/transport/transport.go:393
google.golang.org/grpc.(*csAttempt).recvMsg(0xc0022cca90, {0x2978660, 0xc001dee200}, 0x1000000000000?)
	google.golang.org/grpc@v1.67.1/stream.go:1115 +0x11a
google.golang.org/grpc.(*clientStream).RecvMsg.func1(0x7b?)
	google.golang.org/grpc@v1.67.1/stream.go:969 +0x1f
google.golang.org/grpc.(*clientStream).withRetry(0xc0001b99e0, 0xc00213f1d8, 0xc00213f1c8)
	google.golang.org/grpc@v1.67.1/stream.go:789 +0x13a
google.golang.org/grpc.(*clientStream).RecvMsg(0xc0001b99e0, {0x2978660?, 0xc001dee200?})
	google.golang.org/grpc@v1.67.1/stream.go:968 +0x155
google.golang.org/grpc.invoke({0x33b1690?, 0xc001e24b10?}, {0x2c86b57?, 0x16?}, {0x2a8f480, 0xc001b40180}, {0x2978660, 0xc001dee200}, 0xc001e24510?, {0xc0020fc010, ...})
	google.golang.org/grpc@v1.67.1/call.go:73 +0xcb
github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingClientInterceptor.func1({0x33b16c8, 0xc001e00190}, {0x2c86b57, 0x23}, {0x2a8f480, 0xc001b40180}, {0x2978660, 0xc001dee200}, 0xc0022c6408, 0x2f7c0b8, ...)
	github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/client.go:60 +0x44d
google.golang.org/grpc.(*ClientConn).Invoke(0xc0022c6408, {0x33b16c8?, 0xc001e00190?}, {0x2c86b57?, 0xc001b40180?}, {0x2a8f480?, 0xc001b40180?}, {0x2978660?, 0xc001dee200?}, {0x0, ...})
	google.golang.org/grpc@v1.67.1/call.go:35 +0x205
github.com/pulumi/pulumi/sdk/v3/proto/go.(*converterClient).ConvertProgram(0xc001dd0210, {0x33b16c8, 0xc001e00190}, 0xc001b40180, {0x0, 0x0, 0x0})
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/proto/go/converter_grpc.pb.go:50 +0xc8
github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.(*converter).ConvertProgram(0xc001e24510, {0x33b16c8, 0xc001e00190}, 0xc00226f080)
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/converter_plugin.go:154 +0x244
github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/convert.runConvert({0x33b1770, 0x4a7ed00}, {0x339dcc8, 0x4a7ed00}, {0x33ace60, 0xc00222bd10}, {0xc00228e180, 0x2, 0xc}, {0xc000012194, ...}, ...)
	github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/convert/convert.go:387 +0x1239
github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/convert.NewConvertCmd.func1(0xc00227b208, {0xc00228e180, 0x2, 0xc})
	github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/convert/convert.go:89 +0x290
github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/convert.NewConvertCmd.RunCmdFunc.func2(0x0?, {0xc00228e180?, 0x0?, 0x0?})
	github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/cmd/cmd.go:45 +0x1c
github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/convert.NewConvertCmd.RunCmdFunc.RunFunc.func3(0xc00227b208, {0xc00228e180, 0x2, 0xc})
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/cmdutil/exit.go:99 +0x3b
github.com/spf13/cobra.(*Command).execute(0xc00227b208, {0xc00228e0c0, 0xc, 0xc})
	github.com/spf13/cobra@v1.8.0/command.go:987 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0xc002212008)
	github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.0/command.go:1039
main.main()
	github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/main.go:55 +0x4e

goroutine 8 [select]:
github.com/golang/glog.(*fileSink).flushDaemon(0x4a54f98)
	github.com/golang/glog@v1.2.2/glog_file.go:349 +0xb4
created by github.com/golang/glog.init.1 in goroutine 1
	github.com/golang/glog@v1.2.2/glog_file.go:164 +0x126

goroutine 9 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc000a2e580)
	go.opencensus.io@v0.24.0/stats/view/worker.go:292 +0x9f
created by go.opencensus.io/stats/view.init.0 in goroutine 1
	go.opencensus.io@v0.24.0/stats/view/worker.go:34 +0x8d

goroutine 34 [chan send]:
main.NewPulumiCmd.func1.2()
	github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/pulumi.go:283 +0x39
created by main.NewPulumiCmd.func1 in goroutine 1
	github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/pulumi.go:282 +0x3bb

goroutine 35 [chan receive]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil.serveWithOptions.func1()
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/rpcutil/serve.go:118 +0x2a
created by github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil.serveWithOptions in goroutine 1
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/rpcutil/serve.go:117 +0x7bd

goroutine 36 [IO wait]:
internal/poll.runtime_pollWait(0x7f6b22450ed0, 0x72)
	runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc0021bfb00?, 0xc001dd1930?, 0x0)
	internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0xc0021bfb00)
	internal/poll/fd_unix.go:620 +0x295
net.(*netFD).accept(0xc0021bfb00)
	net/fd_unix.go:172 +0x29
net.(*TCPListener).accept(0xc002295b80)
	net/tcpsock_posix.go:159 +0x1e
net.(*TCPListener).Accept(0xc002295b80)
	net/tcpsock.go:372 +0x30
google.golang.org/grpc.(*Server).Serve(0xc00228b000, {0x33ad130, 0xc002295b80})
	google.golang.org/grpc@v1.67.1/server.go:884 +0x46c
github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil.serveWithOptions.func2()
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/rpcutil/serve.go:128 +0x2c
created by github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil.serveWithOptions in goroutine 1
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/rpcutil/serve.go:127 +0x851

goroutine 37 [chan receive]:
github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.NewDefaultHost.func1()
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/host.go:157 +0x65
created by github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.NewDefaultHost in goroutine 1
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/host.go:156 +0x34f

goroutine 38 [chan receive]:
github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.NewDefaultHost.func2()
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/host.go:165 +0x65
created by github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.NewDefaultHost in goroutine 1
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/host.go:164 +0x391

goroutine 39 [IO wait]:
internal/poll.runtime_pollWait(0x7f6b22450958, 0x72)
	runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc00226ef00?, 0xc0022e0000?, 0x1)
	internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc00226ef00, {0xc0022e0000, 0x1000, 0x1000})
	internal/poll/fd_unix.go:165 +0x27a
os.(*File).read(...)
	os/file_posix.go:29
os.(*File).Read(0xc0021a4970, {0xc0022e0000?, 0x7f6adb8a56a0?, 0x1000?})
	os/file.go:124 +0x52
bufio.(*Reader).fill(0xc001e1cf40)
	bufio/bufio.go:110 +0x103
bufio.(*Reader).ReadSlice(0xc001e1cf40, 0xa)
	bufio/bufio.go:376 +0x29
bufio.(*Reader).collectFragments(0xc001e1cf40, 0xa)
	bufio/bufio.go:451 +0x70
bufio.(*Reader).ReadString(0x0?, 0x0?)
	bufio/bufio.go:498 +0x1f
github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.newPlugin[...].func2(0x2, 0xc0021af7a0?)
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/plugin.go:289 +0x1e7
created by github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.newPlugin[...] in goroutine 1
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/plugin.go:330 +0xa45

goroutine 18 [IO wait]:
internal/poll.runtime_pollWait(0x7f6b22450b88, 0x72)
	runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc00226ee40?, 0xc0022e1000?, 0x1)
	internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc00226ee40, {0xc0022e1000, 0x1000, 0x1000})
	internal/poll/fd_unix.go:165 +0x27a
os.(*File).read(...)
	os/file_posix.go:29
os.(*File).Read(0xc0021a4960, {0xc0022e1000?, 0x20?, 0xc0004fbcc9?})
	os/file.go:124 +0x52
bufio.(*Reader).fill(0xc001e1ff40)
	bufio/bufio.go:110 +0x103
bufio.(*Reader).ReadSlice(0xc001e1ff40, 0xa)
	bufio/bufio.go:376 +0x29
bufio.(*Reader).collectFragments(0xc001e1ff40, 0xa)
	bufio/bufio.go:451 +0x70
bufio.(*Reader).ReadString(0x0?, 0x0?)
	bufio/bufio.go:498 +0x1f
github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.newPlugin[...].func2(0x1, 0xc001fd2000?)
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/plugin.go:289 +0x1e7
created by github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.newPlugin[...] in goroutine 1
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/resource/plugin/plugin.go:371 +0x10c5

goroutine 19 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0020fc240, {0x33b16c8, 0xc002045040})
	google.golang.org/grpc@v1.67.1/internal/grpcsync/callback_serializer.go:88 +0x10c
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 1
	google.golang.org/grpc@v1.67.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 20 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0020fc270, {0x33b16c8, 0xc002045090})
	google.golang.org/grpc@v1.67.1/internal/grpcsync/callback_serializer.go:88 +0x10c
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 1
	google.golang.org/grpc@v1.67.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 21 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0020fc2a0, {0x33b16c8, 0xc0020450e0})
	google.golang.org/grpc@v1.67.1/internal/grpcsync/callback_serializer.go:88 +0x10c
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 1
	google.golang.org/grpc@v1.67.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 42 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc000486040, 0x1)
	google.golang.org/grpc@v1.67.1/internal/transport/controlbuf.go:412 +0x108
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc0021bff00)
	google.golang.org/grpc@v1.67.1/internal/transport/controlbuf.go:575 +0x7b
google.golang.org/grpc/internal/transport.NewServerTransport.func2()
	google.golang.org/grpc@v1.67.1/internal/transport/http2_server.go:335 +0xde
created by google.golang.org/grpc/internal/transport.NewServerTransport in goroutine 41
	google.golang.org/grpc@v1.67.1/internal/transport/http2_server.go:333 +0x18be

goroutine 13 [IO wait]:
internal/poll.runtime_pollWait(0x7f6b22450840, 0x72)
	runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc001b40080?, 0xc001e36000?, 0x0)
	internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc001b40080, {0xc001e36000, 0x8000, 0x8000})
	internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc001b40080, {0xc001e36000?, 0x18?, 0x18?})
	net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc000096518, {0xc001e36000?, 0x800000601?, 0xc000000000?})
	net/net.go:189 +0x45
bufio.(*Reader).Read(0xc00226f020, {0xc0000511c0, 0x9, 0xc000098e08?})
	bufio/bufio.go:241 +0x197
io.ReadAtLeast({0x338be60, 0xc00226f020}, {0xc0000511c0, 0x9, 0x9}, 0x9)
	io/io.go:335 +0x90
io.ReadFull(...)
	io/io.go:354
golang.org/x/net/http2.readFrameHeader({0xc0000511c0, 0x9, 0xc00218b230?}, {0x338be60?, 0xc00226f020?})
	golang.org/x/net@v0.33.0/http2/frame.go:237 +0x65
golang.org/x/net/http2.(*Framer).ReadFrame(0xc000051180)
	golang.org/x/net@v0.33.0/http2/frame.go:501 +0x85
google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc000fb86c8, 0xc002211340)
	google.golang.org/grpc@v1.67.1/internal/transport/http2_client.go:1645 +0x1c6
created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 22
	google.golang.org/grpc@v1.67.1/internal/transport/http2_client.go:412 +0x1e39

goroutine 40 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc001dee100, 0x1)
	google.golang.org/grpc@v1.67.1/internal/transport/controlbuf.go:412 +0x108
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc001fb0200)
	google.golang.org/grpc@v1.67.1/internal/transport/controlbuf.go:575 +0x7b
google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
	google.golang.org/grpc@v1.67.1/internal/transport/http2_client.go:470 +0xd2
created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 22
	google.golang.org/grpc@v1.67.1/internal/transport/http2_client.go:468 +0x241b

goroutine 14 [chan receive]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil.serveWithOptions.func1()
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/rpcutil/serve.go:118 +0x2a
created by github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil.serveWithOptions in goroutine 1
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/rpcutil/serve.go:117 +0x7bd

goroutine 15 [IO wait]:
internal/poll.runtime_pollWait(0x7f6b22450a70, 0x72)
	runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc001b40100?, 0x900000036?, 0x0)
	internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0xc001b40100)
	internal/poll/fd_unix.go:620 +0x295
net.(*netFD).accept(0xc001b40100)
	net/fd_unix.go:172 +0x29
net.(*TCPListener).accept(0xc001dee140)
	net/tcpsock_posix.go:159 +0x1e
net.(*TCPListener).Accept(0xc001dee140)
	net/tcpsock.go:372 +0x30
google.golang.org/grpc.(*Server).Serve(0xc001dea000, {0x33ad130, 0xc001dee140})
	google.golang.org/grpc@v1.67.1/server.go:884 +0x46c
github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil.serveWithOptions.func2()
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/rpcutil/serve.go:128 +0x2c
created by github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil.serveWithOptions in goroutine 1
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/go/common/util/rpcutil/serve.go:127 +0x851

goroutine 43 [select]:
google.golang.org/grpc/internal/transport.(*http2Server).keepalive(0xc001ea21a0)
	google.golang.org/grpc@v1.67.1/internal/transport/http2_server.go:1183 +0x1f3
created by google.golang.org/grpc/internal/transport.NewServerTransport in goroutine 41
	google.golang.org/grpc@v1.67.1/internal/transport/http2_server.go:356 +0x1905

goroutine 44 [IO wait]:
internal/poll.runtime_pollWait(0x7f6b22450db8, 0x72)
	runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc0021bfe00?, 0xc000822000?, 0x0)
	internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc0021bfe00, {0xc000822000, 0x8000, 0x8000})
	internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc0021bfe00, {0xc000822000?, 0x1060100000000?, 0x8?})
	net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc0021a49a8, {0xc000822000?, 0x800010601?, 0x0?})
	net/net.go:189 +0x45
bufio.(*Reader).Read(0xc001de62a0, {0xc00033c040, 0x9, 0xc000098708?})
	bufio/bufio.go:241 +0x197
io.ReadAtLeast({0x338be60, 0xc001de62a0}, {0xc00033c040, 0x9, 0x9}, 0x9)
	io/io.go:335 +0x90
io.ReadFull(...)
	io/io.go:354
golang.org/x/net/http2.readFrameHeader({0xc00033c040, 0x9, 0xc0020d20f0?}, {0x338be60?, 0xc001de62a0?})
	golang.org/x/net@v0.33.0/http2/frame.go:237 +0x65
golang.org/x/net/http2.(*Framer).ReadFrame(0xc00033c000)
	golang.org/x/net@v0.33.0/http2/frame.go:501 +0x85
google.golang.org/grpc/internal/transport.(*http2Server).HandleStreams(0xc001ea21a0, {0x33b1690, 0xc0022ca9c0}, 0xc0022ca9f0)
	google.golang.org/grpc@v1.67.1/internal/transport/http2_server.go:644 +0x107
google.golang.org/grpc.(*Server).serveStreams(0xc001dea000, {0x33b1770?, 0x4a7ed00?}, {0x33beaa0, 0xc001ea21a0}, {0x33bd030?, 0xc0021a49a8?})
	google.golang.org/grpc@v1.67.1/server.go:1023 +0x396
google.golang.org/grpc.(*Server).handleRawConn.func1()
	google.golang.org/grpc@v1.67.1/server.go:958 +0x56
created by google.golang.org/grpc.(*Server).handleRawConn in goroutine 41
	google.golang.org/grpc@v1.67.1/server.go:957 +0x1c6

goroutine 46 [runnable]:
google.golang.org/protobuf/internal/impl.fieldCoder({0x33d5ae8, 0xc002285200}, {0x33e1c18, 0x264c5c0})
	google.golang.org/protobuf@v1.35.1/internal/impl/codec_tables.go:37 +0x32c5
google.golang.org/protobuf/internal/impl.(*MessageInfo).makeCoderMethods(0xc001bc0aa8, {0x33e1c18?, 0x2a0d6c0?}, {0x8, {0x33e1c18, 0x26892e0}, 0xffffffffffffffff, {0x0, 0x0}, 0x10, ...})
	google.golang.org/protobuf@v1.35.1/internal/impl/codec_message.go:116 +0x688
google.golang.org/protobuf/internal/impl.(*MessageInfo).initOnce(0xc001bc0aa8)
	google.golang.org/protobuf@v1.35.1/internal/impl/message.go:91 +0x1f0
google.golang.org/protobuf/internal/impl.(*MessageInfo).init(...)
	google.golang.org/protobuf@v1.35.1/internal/impl/message.go:72
google.golang.org/protobuf/internal/impl.(*messageState).ProtoMethods(0x7f6adb70c250?)
	google.golang.org/protobuf@v1.35.1/internal/impl/message_reflect_gen.go:31 +0x2e
google.golang.org/protobuf/proto.protoMethods(...)
	google.golang.org/protobuf@v1.35.1/proto/proto_methods.go:19
google.golang.org/protobuf/proto.MarshalOptions.size({{}, 0x80?, 0xe5?, 0xde?}, {0x33c7d40, 0xc001dee580})
	google.golang.org/protobuf@v1.35.1/proto/size.go:33 +0x38
google.golang.org/protobuf/proto.MarshalOptions.Size({{}, 0xe0?, 0xa2?, 0x97?}, {0x3391660?, 0xc001dee580?})
	google.golang.org/protobuf@v1.35.1/proto/size.go:26 +0x4c
google.golang.org/protobuf/proto.Size(...)
	google.golang.org/protobuf@v1.35.1/proto/size.go:16
google.golang.org/grpc/encoding/proto.(*codecV2).Marshal(0xc000a0e101?, {0x297a2e0?, 0xc001dee580?})
	google.golang.org/grpc@v1.67.1/encoding/proto/proto.go:49 +0x7a
google.golang.org/grpc.encode({0x7f6adb8a9770?, 0x4a7ed00?}, {0x297a2e0?, 0xc001dee580?})
	google.golang.org/grpc@v1.67.1/rpc_util.go:694 +0x4a
google.golang.org/grpc.(*Server).sendResponse(0xc001dea000, {0x33b1690, 0xc001e251d0}, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef200, {0x297a2e0, 0xc001dee580}, {0x0, 0x0}, ...)
	google.golang.org/grpc@v1.67.1/server.go:1140 +0xab
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001dea000, {0x33b1690, 0xc001e25140}, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef200, 0xc001e24750, 0x49cc0f0, 0x0)
	google.golang.org/grpc@v1.67.1/server.go:1441 +0xf7d
google.golang.org/grpc.(*Server).handleStream(0xc001dea000, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef200)
	google.golang.org/grpc@v1.67.1/server.go:1805 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.67.1/server.go:1029 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 44
	google.golang.org/grpc@v1.67.1/server.go:1040 +0x125

goroutine 47 [sync.Mutex.Lock]:
sync.runtime_SemacquireMutex(0xc0020c66b0?, 0x18?, 0xd1eba5?)
	runtime/sema.go:95 +0x25
sync.(*Mutex).lockSlow(0xc001bc0ae8)
	sync/mutex.go:173 +0x15d
sync.(*Mutex).Lock(...)
	sync/mutex.go:92
google.golang.org/protobuf/internal/impl.(*MessageInfo).initOnce(0xc001bc0aa8)
	google.golang.org/protobuf@v1.35.1/internal/impl/message.go:77 +0x65
google.golang.org/protobuf/internal/impl.(*MessageInfo).init(...)
	google.golang.org/protobuf@v1.35.1/internal/impl/message.go:72
google.golang.org/protobuf/internal/impl.(*messageState).ProtoMethods(0x10?)
	google.golang.org/protobuf@v1.35.1/internal/impl/message_reflect_gen.go:31 +0x2e
google.golang.org/protobuf/proto.protoMethods(...)
	google.golang.org/protobuf@v1.35.1/proto/proto_methods.go:19
google.golang.org/protobuf/proto.MarshalOptions.size({{}, 0x40?, 0x44?, 0x6?}, {0x33c7d40, 0xc001064440})
	google.golang.org/protobuf@v1.35.1/proto/size.go:33 +0x38
google.golang.org/protobuf/proto.MarshalOptions.Size({{}, 0xe0?, 0xa2?, 0x97?}, {0x3391660?, 0xc001064440?})
	google.golang.org/protobuf@v1.35.1/proto/size.go:26 +0x4c
google.golang.org/protobuf/proto.Size(...)
	google.golang.org/protobuf@v1.35.1/proto/size.go:16
google.golang.org/grpc/encoding/proto.(*codecV2).Marshal(0xc000a0e101?, {0x297a2e0?, 0xc001064440?})
	google.golang.org/grpc@v1.67.1/encoding/proto/proto.go:49 +0x7a
google.golang.org/grpc.encode({0x7f6adb8a9770?, 0x4a7ed00?}, {0x297a2e0?, 0xc001064440?})
	google.golang.org/grpc@v1.67.1/rpc_util.go:694 +0x4a
google.golang.org/grpc.(*Server).sendResponse(0xc001dea000, {0x33b1690, 0xc0021409c0}, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef320, {0x297a2e0, 0xc001064440}, {0x0, 0x0}, ...)
	google.golang.org/grpc@v1.67.1/server.go:1140 +0xab
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001dea000, {0x33b1690, 0xc002140930}, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef320, 0xc001e24750, 0x49cc0f0, 0x0)
	google.golang.org/grpc@v1.67.1/server.go:1441 +0xf7d
google.golang.org/grpc.(*Server).handleStream(0xc001dea000, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef320)
	google.golang.org/grpc@v1.67.1/server.go:1805 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.67.1/server.go:1029 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 44
	google.golang.org/grpc@v1.67.1/server.go:1040 +0x125

goroutine 48 [runnable]:
sync.runtime_SemacquireMutex(0x18?, 0xd8?, 0xc0009c1488?)
	runtime/sema.go:95 +0x25
sync.(*Mutex).lockSlow(0xc001bc0848)
	sync/mutex.go:173 +0x15d
sync.(*Mutex).Lock(...)
	sync/mutex.go:92
google.golang.org/protobuf/internal/impl.(*MessageInfo).initOnce(0xc001bc0808)
	google.golang.org/protobuf@v1.35.1/internal/impl/message.go:77 +0x65
google.golang.org/protobuf/internal/impl.(*MessageInfo).init(...)
	google.golang.org/protobuf@v1.35.1/internal/impl/message.go:72
google.golang.org/protobuf/internal/impl.(*messageState).ProtoMethods(0x3391680?)
	google.golang.org/protobuf@v1.35.1/internal/impl/message_reflect_gen.go:31 +0x2e
google.golang.org/protobuf/proto.protoMethods(...)
	google.golang.org/protobuf@v1.35.1/proto/proto_methods.go:19
google.golang.org/protobuf/proto.UnmarshalOptions.unmarshal({{}, 0x1, 0x1, 0x0, {0x33a04a8, 0xc0001e47e0}, 0x2710}, {0xc00218b2dd, 0xd, 0xd}, ...)
	google.golang.org/protobuf@v1.35.1/proto/decode.go:95 +0xe2
google.golang.org/protobuf/proto.Unmarshal({0xc00218b2dd, 0xd, 0xd}, {0x3391680?, 0xc000192aa0?})
	google.golang.org/protobuf@v1.35.1/proto/decode.go:57 +0x5d
google.golang.org/grpc/encoding/proto.(*codecV2).Unmarshal(0x7f6b225e5188?, {0xc0005e0130?, 0x1?, 0x1?}, {0x2a30860?, 0xc000192aa0?})
	google.golang.org/grpc@v1.67.1/encoding/proto/proto.go:80 +0xfa
google.golang.org/grpc.(*Server).processUnaryRPC.func2({0x2a30860, 0xc000192aa0})
	google.golang.org/grpc@v1.67.1/server.go:1367 +0xe2
github.com/pulumi/pulumi/sdk/v3/proto/go/codegen._Mapper_GetMapping_Handler({0x280d400, 0xc000a0e100}, {0x33b1690, 0xc0009960c0}, 0xc000322a80, 0xc000a0e140)
	github.com/pulumi/pulumi/sdk/v3@v3.151.0/proto/go/codegen/mapper_grpc.pb.go:85 +0x55
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001dea000, {0x33b1690, 0xc000996030}, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef440, 0xc001e24750, 0x49cc0f0, 0x0)
	google.golang.org/grpc@v1.67.1/server.go:1394 +0xe2b
google.golang.org/grpc.(*Server).handleStream(0xc001dea000, {0x33beaa0, 0xc001ea21a0}, 0xc0021ef440)
	google.golang.org/grpc@v1.67.1/server.go:1805 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.67.1/server.go:1029 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 44
	google.golang.org/grpc@v1.67.1/server.go:1040 +0x125




goroutine 1 [running]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.152.0/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.AssertNoErrorf({0x2584dc0, 0xc001214600}, {0x1e69a18?, 0x4d706c?}, {0x0?, 0xc0000d01b0?, 0x1e40c4e?})
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.152.0/go/common/util/contract/assert.go:51 +0x12b
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*cliConverter).FinishConvertingExamples(_, {{0xc00084cbe0, 0x9}, {0x1e37a33, 0x9}, {0x2592180, 0x1f}, {0x1eb43ed, 0x3f}, {0xc000786800, ...}, ...})
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.103.1-0.20250227182354-ebf4276b8dd1/pkg/tfgen/convert_cli.go:156 +0x107
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*Generator).convertExamplesInSchema(_, {{0xc00084cbe0, 0x9}, {0x1e37a33, 0x9}, {0x2592180, 0x1f}, {0x1eb43ed, 0x3f}, {0xc000786800, ...}, ...})
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.103.1-0.20250227182354-ebf4276b8dd1/pkg/tfgen/generate_schema.go:1116 +0x8dd
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*Generator).UnstableGenerateFromSchema(0xc000905808, 0x259e780?)
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.103.1-0.20250227182354-ebf4276b8dd1/pkg/tfgen/generate.go:996 +0x1be
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*Generator).Generate(0xc000905808)
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.103.1-0.20250227182354-ebf4276b8dd1/pkg/tfgen/generate.go:939 +0x45
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.Main.func1({{0x1e37988, 0x9}, {0x2592180, 0x1f}, {0x7fffa724dba7, 0x6}, {{0x25bace0, 0xc0007856b0}, {0x1e37988, 0x9}, ...}, ...})
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.103.1-0.20250227182354-ebf4276b8dd1/pkg/tfgen/main.go:52 +0x59
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.newTFGenCmd.func1(0x0?, {0xc0007ec6c0, 0x1, 0x0?})
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.103.1-0.20250227182354-ebf4276b8dd1/pkg/tfgen/main.go:164 +0x5fc
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.newTFGenCmd.RunFunc.func4(0xc000905208, {0xc0007ec6c0, 0x1, 0x3})
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.152.0/go/common/util/cmdutil/exit.go:99 +0x3b
github.com/spf13/cobra.(*Command).execute(0xc000905208, {0xc000052090, 0x3, 0x3})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987 +0xab1
github.com/spf13/cobra.(*Command).ExecuteC(0xc000905208)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.MainWithCustomGenerate({_, _}, {_, _}, {{0x25bace0, 0xc0007856b0}, {0x1e37988, 0x9}, {0x0, 0x0}, ...}, ...)
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.103.1-0.20250227182354-ebf4276b8dd1/pkg/tfgen/main.go:61 +0x6f
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.Main({_, _}, {_, _}, {{0x25bace0, 0xc0007856b0}, {0x1e37988, 0x9}, {0x0, 0x0}, ...})
	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.103.1-0.20250227182354-ebf4276b8dd1/pkg/tfgen/main.go:44 +0x11f
main.main()
	/home/runner/work/pulumi-cloudamqp/pulumi-cloudamqp/provider/cmd/pulumi-tfgen-cloudamqp/main.go:26 +0xa7
make: *** [Makefile:250: .make/schema] Error 2
```

### Output of `pulumi about`

N/A

### Additional context

N/A

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/changelog/pending/20250227--programgen--fix-concurrent-map-write-panics-in-the-caching-mapper.yaml
+++ b/changelog/pending/20250227--programgen--fix-concurrent-map-write-panics-in-the-caching-mapper.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: fix
+  scope: programgen
+  description: Fix concurrent map write panics in the caching mapper

--- a/pkg/codegen/convert/caching_mapper.go
+++ b/pkg/codegen/convert/caching_mapper.go
@@ -14,7 +14,10 @@
 
 package convert
 
-import "context"
+import (
+	"context"
+	"sync"
+)
 
 // cachingMapper wraps another Mapper, caching the results of GetMapping calls by source provider name.
 type cachingMapper struct {
@@ -23,6 +26,9 @@ type cachingMapper struct {
 
 	// A cache of provider mappings, keyed by source provider name.
 	entries map[string][]byte
+
+	// Mutex to protect concurrent access to the entries map
+	mu sync.RWMutex
 }
 
 // NewCachingMapper creates a new caching mapper backed by the given Mapper.
@@ -39,7 +45,11 @@ func (m *cachingMapper) GetMapping(
 	provider string,
 	hint *MapperPackageHint,
 ) ([]byte, error) {
-	if mapping, ok := m.entries[provider]; ok {
+	m.mu.RLock()
+	mapping, ok := m.entries[provider]
+	m.mu.RUnlock()
+
+	if ok {
 		return mapping, nil
 	}
 
@@ -48,6 +58,9 @@ func (m *cachingMapper) GetMapping(
 		return nil, err
 	}
 
+	m.mu.Lock()
 	m.entries[provider] = mapping
+	m.mu.Unlock()
+
 	return mapping, nil
 }

--- a/pkg/codegen/convert/caching_mapper_test.go
+++ b/pkg/codegen/convert/caching_mapper_test.go
@@ -16,6 +16,7 @@ package convert
 
 import (
 	"context"
+	"sync"
 	"testing"
 
 	"github.com/blang/semver"
@@ -76,4 +77,66 @@ func TestCachingPluginMapper_OnlyInstallsOnce(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, []byte{}, data)
 	assert.Equal(t, 1, installCalled, "install should only be called once when a caching mapper is used")
+}
+
+// TestCachingPluginMapper_ConcurrentAccess tests that the caching mapper correctly
+// handles concurrent access from multiple goroutines.
+func TestCachingPluginMapper_ConcurrentAccess(t *testing.T) {
+	t.Parallel()
+
+	// Arrange.
+	ws := &testWorkspace{}
+
+	providerFactory := func(descriptor workspace.PackageDescriptor) (plugin.Provider, error) {
+		t.Fatal("should not be called")
+		return nil, nil
+	}
+
+	installPlugin := func(pluginName string) *semver.Version {
+		return nil
+	}
+
+	baseMapper, err := NewBasePluginMapper(
+		ws,
+		"key", /*conversionKey*/
+		providerFactory,
+		installPlugin,
+		nil, /*mappings*/
+	)
+	mapper := NewCachingMapper(baseMapper)
+	assert.NoError(t, err)
+	assert.NotNil(t, mapper)
+
+	// Act.
+	// Run multiple goroutines concurrently that all try to get mappings
+	// for different providers to maximize map writes
+	const numGoroutines = 100
+
+	// Used to wait for all goroutines to complete
+	var wg sync.WaitGroup
+	wg.Add(numGoroutines)
+
+	// Create multiple goroutines that will all try to access the map at the same time
+	for i := 0; i < numGoroutines; i++ {
+		// Use two different provider names to enforce multiple map writes
+		provider := "gcp"
+		if i%2 == 0 {
+			provider = "aws"
+		}
+
+		go func(p string) {
+			defer wg.Done()
+
+			// Get the mapping - this will cause concurrent map writes without proper locking
+			_, err := mapper.GetMapping(context.Background(), p, nil /*hint*/)
+			assert.NoError(t, err)
+		}(provider)
+	}
+
+	// Wait for all goroutines to complete
+	wg.Wait()
+
+	// Assert.
+	// This test will fail with "concurrent map writes" without proper locking
+	// If it reaches here, the test passes
 }

